dataprofiler/profilers/float_column_profile.py

```diff
@@ -262,15 +266,21 @@ def _update_precision(self, df_series, prev_dependent_properties,
             self.precision['mean'] = self.precision['sum'] \
                 / self.precision['sample_size']
 
-        # Calculated outside        
+        # Calculated outside
         self.precision['std'] = math.sqrt(self.precision['var'])
 
         # Margin of error, 99.9% confidence level
         self.precision['margin_of_error'] = round(
             self.__z_value_precision * (
                 self.precision['std'] / math.sqrt(self.precision['sample_size'])
             ), int(self.precision['max'])
-        )            
+        )
+
+        # Set the significant figures
+        sigfigs = int(self.precision['max'])
+        for key in ['mean', 'var', 'std', 'margin_of_error']:
+            self.precision[key] = \
+                float('{:.{p}g}'.format(self.precision[key], p=sigfigs))
```

Comment:
I'm not sure why, but this incorrectly rounds: `float('{:.{p}g}'.format(.15, p=1))` out: `0.1` expected: `0.2`
what's weird is `float('{:.{p}g}'.format(15, p=1))` is correct and becomes `20`. Not sure diff

Reply:
There are no good rounding functions...

```
>>> float('{:.{p}g}'.format(.15, p=1))
0.1
>>> float('{:.{p}g}'.format(.155, p=1))
0.2
>>> float('{:.{p}g}'.format(15, p=1))
20.0
>>> float('{:.{p}g}'.format(25, p=1))
20.0
>>> float('{:.{p}g}'.format(35, p=1))
40.0
```

Attempted an alternative approach:
```
from math import log10, floor
>>> def round_sig(x, sig=99):
...     return round(x, sig-int(floor(log10(abs(x))))-1)
... 
>>> round_sig(0.0232)
0.0232
>>> round_sig(0.0232, sig=1)
0.02
>>> round_sig(0.15, sig=1)
0.1
>>> round_sig(0.155, sig=1)
0.2
>>> round_sig(0.151, sig=1)
0.2
>>> round_sig(15, sig=1)
20
>>> round_sig(35, sig=1)
40
``

Reply:
Think this is the issue, just `surprised` - https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html 

In the scheme of things I believe this is a minimal issue. Numbers under zero probably aren't going to happen (where the issue primarily arises) and often we wont see 15 sigfigs. 5 rounds appropriately via their current method(s).